virttest/utils_test/__init__.py

```diff
@@ -718,6 +718,165 @@ def transfer_data(session, host_cmd, guest_cmd, n_time, timeout,
         session.close()
 
 
+def run_avocado(vm, params, test, testlist=[], timeout=3600,
+                testrepo="https://github.com/avocado-framework-tests/avocado-misc-tests.git",
+                installtype="pip", install=False, add_args=None, ignore_result=True):
+    """
+    pip, package, git
+
+    :param vm: VM object
+    :param params: VM param
+    :param testlist: testlist as list of tuples like (testcase, muxfile)
+    :param timeout: test timeout
+    :param testrepo: test repository default is avocado-misc-tests
+    :param installtype: how to install avocado, supported types
+                        pip, package, git
+    :param install: flag to reinstall incase of avocado present inside guest
+    :param add_args: additional args to be passed to the avocado cmdline
+    :param ignore_result:
+    """
+    plugins = {'pip': ['avocado-framework-plugin-varianter-yaml-to-mux', 'avocado-framework-plugin-result-html'],
+               'package': [],
+               'git': ['varianter_yaml_to_mux', 'html']}
+    try:
+        session = vm.wait_for_login()
+    except:
+        logging.error("Unable to get VM session, skipped to run avocado test")
+        return False
```

Comment:
@balamuruhans @luckyh Got it, as this is going to be setup time issue, good to fail out here, will do it.